src/modules/credential/userpass/path_users.rs

```diff
@@ -0,0 +1,251 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use serde::{Deserialize, Serialize};
+
+use super::{UserPassBackend, UserPassBackendInner};
+use crate::{
+    errors::RvError,
+    logical::{Backend, Field, FieldType, Operation, Path, PathOperation, Request, Response},
+    new_fields, new_fields_internal, new_path, new_path_internal,
+    storage::StorageEntry,
+    utils::{deserialize_duration, serialize_duration},
+};
+
+//const DEFAULT_MAX_TTL: Duration = Duration::from_secs(365*24*60*60 as u64);
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct UserEntry {
+    pub password_hash: String,
+    pub policies: Vec<String>,
+    #[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
+    pub ttl: Duration,
+    #[serde(serialize_with = "serialize_duration", deserialize_with = "deserialize_duration")]
+    pub max_ttl: Duration,
+}
+
+impl Default for UserEntry {
+    fn default() -> Self {
+        Self {
+            password_hash: String::new(),
+            policies: Vec::new(),
+            ttl: Duration::from_secs(0),
+            max_ttl: Duration::from_secs(0),
+        }
+    }
+}
+
+impl UserPassBackend {
+    pub fn users_path(&self) -> Path {
+        let userpass_backend_ref1 = Arc::clone(&self.inner);
+        let userpass_backend_ref2 = Arc::clone(&self.inner);
+        let userpass_backend_ref3 = Arc::clone(&self.inner);
+
+        let path = new_path!({
+            pattern: r"users/(?P<username>\w[\w-]+\w)",
+            fields: {
+                "username": {
+                    field_type: FieldType::Str,
+                    required: true,
+                    description: "Username of the user."
+                },
+                "password": {
+                    field_type: FieldType::Str,
```

Comment:
Same as previous comment.

Reply:
> Same as previous comment.

Added FieldType::SecretStr type, which will be removed from the Map after the request handling function is executed. The parameter memory will be reclaimed by Rust, but whether the sensitive string remains in memory depends on when that memory is overwritten next.

Reply:
Why not clear the memory by writing in enough zeros before it's removed from the Map? Or do you mean if the field has been removed from the map, then no one can easily find out the address of the field, so the sensitive data will not leaked?

Reply:
> Why not clear the memory by writing in enough zeros before it's removed from the Map? Or do you mean if the field has been removed from the map, then no one can easily find out the address of the field, so the sensitive data will not leaked?

Since serde_json::Value does not provide a mutable reference/interface to modify its internal memory directly, it is only possible to replace the value of its elements. However, this should merely change the pointer of the element to point to the new value, and the original value would be released, whether the sensitive data is overwritten depends on when Rust reuses that section of memory next time.

Reply:
You should leave a line of comment in the code like:

~~~
//TODO: the sensitive data is still in the memory. Need to totally resolve this in `serde_json` someday.
~~~

Reply:
> You should leave a line of comment in the code like:
> 
> ```
> //TODO: the sensitive data is still in the memory. Need to totally resolve this in `serde_json` someday.
> ```
done：
<img width="830" alt="image" src="https://github.com/Tongsuo-Project/RustyVault/assets/7484739/ab81b935-a6d3-46f5-bc69-834c53b0736e">